src/consumer/runner.js

```diff
@@ -312,56 +312,48 @@ module.exports = class Runner {
       return
     }
 
-    const enqueuedTasks = []
     let expectedNumberOfExecutions = 0
     let numberOfExecutions = 0
     const { lock, unlock, unlockWithError } = barrier()
 
     while (true) {
-      const result = iterator.next()
+      const result = await iterator.next()
```

Comment:
I was abit concerned about awaiting in the while loop as it is not going to wait for this and will just loop and saturate cpu... (this is just assumption have not validate that...)

Reply:
The await-ing here is needed because the generator is now doing funny async stuff as well. It shouldn't be a busy loop though, but rather it awaits here for the next promise in the bufferedasynciterator to resolve.